xml/mysql_banners.xml

```diff
@@ -1071,7 +1071,7 @@
     <param pos="0" name="service.product" value="MySQL"/>
   </fingerprint>
 
-  <fingerprint pattern="^(\d{1,2}\.\d{1,3}\.\h{1,3})(?:[^+]+)?\+tld\d">
+  <fingerprint pattern="^(\d{1,2}\.\d{1,3}\.\h{1,3})([^+]+)?\+tld\d">
```

Comment:
Removing the `?:` breaks the unit tests. The unit tests require that the number of capturing regular expressions match the number of uses of those extractions in the params.  This is because in the past we've had problems with over capturing causing memory problems in extreme cases (`(.*)$`, for example), or under capturing.